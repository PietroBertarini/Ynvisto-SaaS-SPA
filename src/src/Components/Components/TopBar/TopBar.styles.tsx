import {
  createStyles, fade, makeStyles,
} from '@material-ui/core/styles';
import styled from '@material-ui/core/styles/styled';
import { AppBar } from '@material-ui/core';
import theme from '../../../Theme/theme';

export const useStyles = makeStyles(() => createStyles({
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    fontSize: 30,
    fontFamily: 'Zurich',
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  text: {
    fontSize: 20,
    color: '#FFFFFF',
    fontFamily: 'SF Pro',
    margin: 'auto',
  },
  icon: {
    fontSize: 24,
    margin: 'auto',
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginRight: theme.spacing(2),
    margin: 'auto',
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(3),
      width: 'auto',
    },
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: 'width 500ms',
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: '15ch',
    },
    '&:hover': {
      width: '30ch',
    },
  },
  sectionDesktop: {
    display: 'none',
    [theme.breakpoints.up('md')]: {
      display: 'flex',
    },
  },
}));

export const StyledAppBar = styled(AppBar)({
  zIndex: theme.zIndex.drawer + 1,
});
