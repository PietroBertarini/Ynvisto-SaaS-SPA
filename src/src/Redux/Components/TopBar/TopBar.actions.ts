import {
  ETopBarTypes, IChangeActivePeriod, IChangeActiveWallet,
} from './TopBar.types';

export const changeActivePeriod = (period: string) : IChangeActivePeriod => ({
  type: ETopBarTypes.CHANGE_ACTIVE_PERIOD,
  payload: period,
});

export const changeActiveWallet = (wallet: string) : IChangeActiveWallet => ({
  type: ETopBarTypes.CHANGE_ACTIVE_WALLET,
  payload: wallet,
});
