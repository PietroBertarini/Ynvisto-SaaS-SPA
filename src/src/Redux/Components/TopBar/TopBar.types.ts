export enum ETopBarTypes {
  CHANGE_ACTIVE_PERIOD = 'CHANGE_ACTIVE_PERIOD',
  CHANGE_ACTIVE_WALLET = 'CHANGE_ACTIVE_WALLET',
}

export interface IChangeActivePeriod {
  type: typeof ETopBarTypes.CHANGE_ACTIVE_PERIOD
  payload: string
}

export interface IChangeActiveWallet {
  type: typeof ETopBarTypes.CHANGE_ACTIVE_WALLET
  payload: string
}
export type TTopBarActionTypes =
    IChangeActivePeriod |
    IChangeActiveWallet

export interface ITopBarState {
  notifications: string[],
  wallets: string[],
  activeWallet: string,
  activePeriod: string,
}
export enum EPeriodType {
  DAILY,
  WEEK,
  MONTH,
  YEAR,
}
