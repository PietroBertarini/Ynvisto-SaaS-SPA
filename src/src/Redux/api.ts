import axios, { AxiosResponse } from 'axios';

/**
 * Localhost YnvistoGateway api.
 */
const api = axios.create({
  baseURL: process.env.REACT_APP_API_ENDPOINT,
});

// /**
//  * Check if the one error have status 401.
//  * @param err: Caught error
//  */
// const isAuthError = (err: any): boolean => {
//   const axiosError = err as AxiosError;
//   // Whenever the erro is not from axios, we cannot try to repair it
//   if (!axiosError?.response) {
//     return false;
//   }
//
//   // Gets the error code from the axios response
//   const httpStatusCode = axiosError.response?.status;
//
//   // If the error code is Unauthorized, mark the error as Unauthorized error
//   return httpStatusCode === 401;
// };

/**
 * Always that callSafely catch error with status 401, will refresh
 * api token and call one more time.(Just 2 times), if this error occurs
 * again, than throw this error.
 * @param apiCall: Api with one method(get,post...)
 * @param authRetry: boolean to check if error 401 was caught one time.
 */
export function* callSafely<T>(apiCall: Promise<AxiosResponse<any>>, authRetry = false) {
  try {
    return yield apiCall;
  } catch (e) {
    if (!authRetry) {
      // TODO: Refresh token
      // yield callSafely(apiCall, true);
    }
    throw (e);
  }
}

export default api;
