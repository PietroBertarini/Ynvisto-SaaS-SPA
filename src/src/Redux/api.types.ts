export enum ApiErrorCode {
  BrokerNoteImportingError = 'BrokerNoteImportingError',
  BrokerNoteImportingWarning = 'BrokerNoteImportingWarning',
  UnknownError = 'UnknownError',
}

export interface IApiError extends IApiErrorModel {
  httpStatusCode?: number,
}

export interface IApiErrorModel{
  errorMessage: string,
  errorCode: ApiErrorCode
}
