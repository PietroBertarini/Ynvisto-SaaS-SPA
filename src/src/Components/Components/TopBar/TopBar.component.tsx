import React from 'react';
import Toolbar from '@material-ui/core/Toolbar';
import { connect, ConnectedProps } from 'react-redux';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import Badge from '@material-ui/core/Badge';
import NotificationsIcon from '@material-ui/icons/Notifications';
import { StyledAppBar, useStyles } from './TopBar.styles';
import { State } from '../../../Redux/root-reducer';

/**
 * Get the stores states that TopBar uses and change.
 * @param state: the stored stated of the Ynvisto.
 */
const mapStateToProps = (state: State) => ({
  activePeriod: state.topBar.activePeriod,
  activeWallet: state.topBar.activeWallet,
  wallets: state.topBar.wallets,
  notifications: state.topBar.notifications,
});
const connector = connect(mapStateToProps);
type PropsFromRedux = ConnectedProps<typeof connector>
export type OwnProps = PropsFromRedux

/**
 * Is the TopBar of Ynvisto that changes the Active Wallet, Active Period,
 * see the notifications, perfil and the search bar.
 * @param props: all states from store that topBar view and change.
 */
const TopBar = (props: OwnProps) => {
  const classes = useStyles();
  return (
    <div className={classes.grow}>
      <StyledAppBar position="fixed">
        <Toolbar>
          <Typography className={classes.title} noWrap>
            Ynvisto
          </Typography>
          <div className={classes.grow} />
          <div className={classes.sectionDesktop}>
            <IconButton aria-label="show 17 new notifications" color="inherit">
              <Badge badgeContent={1} color="secondary">
                <NotificationsIcon />
              </Badge>
            </IconButton>
          </div>
        </Toolbar>
      </StyledAppBar>
    </div>
  );
};

export default connector(TopBar);
