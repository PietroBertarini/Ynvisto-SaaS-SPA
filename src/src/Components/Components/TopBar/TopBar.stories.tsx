import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import TopBar, { OwnProps } from './TopBar.component';

export default {
  title: 'Components/TopBar',
  component: TopBar,
} as Meta;

const Template: Story<OwnProps> = (args) => (<TopBar {...args} />
);

export const TopBarDefault = Template.bind({});
