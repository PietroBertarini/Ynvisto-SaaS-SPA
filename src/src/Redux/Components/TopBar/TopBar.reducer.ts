import { TTopBarActionTypes, ITopBarState, ETopBarTypes } from './TopBar.types';

const INITIAL_STATE : ITopBarState = {
  notifications: [],
  wallets: [],
  activeWallet: 'Todos',
  activePeriod: 'Dia',
};

export default function topBarReducer(
  state: ITopBarState = INITIAL_STATE, action: TTopBarActionTypes,
) {
  switch (action.type) {
    case ETopBarTypes.CHANGE_ACTIVE_PERIOD:
      return {
        ...state,
        activePeriod: action.payload,
      };
    case ETopBarTypes.CHANGE_ACTIVE_WALLET:
      return {
        ...state,
        activeWallet: action.payload,
      };
    default:
      return state;
  }
}
