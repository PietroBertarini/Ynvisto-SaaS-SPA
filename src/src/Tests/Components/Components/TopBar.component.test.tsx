import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import TopBar from '../../../Components/Components/TopBar/TopBar.component';
import { store } from '../../../Redux/store';

export const createMockStore = () => ({
  ...store,
  persistor: {
    persist: () => null,
  },
});

describe('Renders TopBar component and match to previously snapshot', () => {
  let wrapper: any;
  const mockStore = createMockStore();

  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <Provider store={mockStore}>
          <TopBar />
        </Provider>
      </BrowserRouter>,
    );
  });

  it('should render TopBar component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
