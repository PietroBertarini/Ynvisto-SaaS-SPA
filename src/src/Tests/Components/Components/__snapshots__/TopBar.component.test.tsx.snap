// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TopBar component and match to previously snapshot should render TopBar component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "persistor": Object {
            "persist": [Function],
          },
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(TopBar)>
        <TopBar
          activePeriod="Dia"
          activeWallet="Todos"
          dispatch={[Function]}
          notifications={Array []}
          wallets={Array []}
        >
          <div
            className="makeStyles-grow-1"
          >
            <Styled(WithStyles(ForwardRef(AppBar)))
              position="fixed"
            >
              <WithStyles(ForwardRef(AppBar))
                className="WithStyles(ForwardRef(AppBar))-root-11"
                position="fixed"
              >
                <ForwardRef(AppBar)
                  className="WithStyles(ForwardRef(AppBar))-root-11"
                  classes={
                    Object {
                      "colorDefault": "MuiAppBar-colorDefault",
                      "colorInherit": "MuiAppBar-colorInherit",
                      "colorPrimary": "MuiAppBar-colorPrimary",
                      "colorSecondary": "MuiAppBar-colorSecondary",
                      "colorTransparent": "MuiAppBar-colorTransparent",
                      "positionAbsolute": "MuiAppBar-positionAbsolute",
                      "positionFixed": "MuiAppBar-positionFixed",
                      "positionRelative": "MuiAppBar-positionRelative",
                      "positionStatic": "MuiAppBar-positionStatic",
                      "positionSticky": "MuiAppBar-positionSticky",
                      "root": "MuiAppBar-root",
                    }
                  }
                  position="fixed"
                >
                  <WithStyles(ForwardRef(Paper))
                    className="MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary WithStyles(ForwardRef(AppBar))-root-11 mui-fixed"
                    component="header"
                    elevation={4}
                    square={true}
                  >
                    <ForwardRef(Paper)
                      className="MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary WithStyles(ForwardRef(AppBar))-root-11 mui-fixed"
                      classes={
                        Object {
                          "elevation0": "MuiPaper-elevation0",
                          "elevation1": "MuiPaper-elevation1",
                          "elevation10": "MuiPaper-elevation10",
                          "elevation11": "MuiPaper-elevation11",
                          "elevation12": "MuiPaper-elevation12",
                          "elevation13": "MuiPaper-elevation13",
                          "elevation14": "MuiPaper-elevation14",
                          "elevation15": "MuiPaper-elevation15",
                          "elevation16": "MuiPaper-elevation16",
                          "elevation17": "MuiPaper-elevation17",
                          "elevation18": "MuiPaper-elevation18",
                          "elevation19": "MuiPaper-elevation19",
                          "elevation2": "MuiPaper-elevation2",
                          "elevation20": "MuiPaper-elevation20",
                          "elevation21": "MuiPaper-elevation21",
                          "elevation22": "MuiPaper-elevation22",
                          "elevation23": "MuiPaper-elevation23",
                          "elevation24": "MuiPaper-elevation24",
                          "elevation3": "MuiPaper-elevation3",
                          "elevation4": "MuiPaper-elevation4",
                          "elevation5": "MuiPaper-elevation5",
                          "elevation6": "MuiPaper-elevation6",
                          "elevation7": "MuiPaper-elevation7",
                          "elevation8": "MuiPaper-elevation8",
                          "elevation9": "MuiPaper-elevation9",
                          "outlined": "MuiPaper-outlined",
                          "root": "MuiPaper-root",
                          "rounded": "MuiPaper-rounded",
                        }
                      }
                      component="header"
                      elevation={4}
                      square={true}
                    >
                      <header
                        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary WithStyles(ForwardRef(AppBar))-root-11 mui-fixed MuiPaper-elevation4"
                      >
                        <WithStyles(ForwardRef(Toolbar))>
                          <ForwardRef(Toolbar)
                            classes={
                              Object {
                                "dense": "MuiToolbar-dense",
                                "gutters": "MuiToolbar-gutters",
                                "regular": "MuiToolbar-regular",
                                "root": "MuiToolbar-root",
                              }
                            }
                          >
                            <div
                              className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
                            >
                              <WithStyles(ForwardRef(Typography))
                                className="makeStyles-title-3"
                                noWrap={true}
                              >
                                <ForwardRef(Typography)
                                  className="makeStyles-title-3"
                                  classes={
                                    Object {
                                      "alignCenter": "MuiTypography-alignCenter",
                                      "alignJustify": "MuiTypography-alignJustify",
                                      "alignLeft": "MuiTypography-alignLeft",
                                      "alignRight": "MuiTypography-alignRight",
                                      "body1": "MuiTypography-body1",
                                      "body2": "MuiTypography-body2",
                                      "button": "MuiTypography-button",
                                      "caption": "MuiTypography-caption",
                                      "colorError": "MuiTypography-colorError",
                                      "colorInherit": "MuiTypography-colorInherit",
                                      "colorPrimary": "MuiTypography-colorPrimary",
                                      "colorSecondary": "MuiTypography-colorSecondary",
                                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                      "displayBlock": "MuiTypography-displayBlock",
                                      "displayInline": "MuiTypography-displayInline",
                                      "gutterBottom": "MuiTypography-gutterBottom",
                                      "h1": "MuiTypography-h1",
                                      "h2": "MuiTypography-h2",
                                      "h3": "MuiTypography-h3",
                                      "h4": "MuiTypography-h4",
                                      "h5": "MuiTypography-h5",
                                      "h6": "MuiTypography-h6",
                                      "noWrap": "MuiTypography-noWrap",
                                      "overline": "MuiTypography-overline",
                                      "paragraph": "MuiTypography-paragraph",
                                      "root": "MuiTypography-root",
                                      "srOnly": "MuiTypography-srOnly",
                                      "subtitle1": "MuiTypography-subtitle1",
                                      "subtitle2": "MuiTypography-subtitle2",
                                    }
                                  }
                                  noWrap={true}
                                >
                                  <p
                                    className="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-noWrap"
                                  >
                                    Ynvisto
                                  </p>
                                </ForwardRef(Typography)>
                              </WithStyles(ForwardRef(Typography))>
                              <div
                                className="makeStyles-grow-1"
                              />
                              <div
                                className="makeStyles-sectionDesktop-10"
                              >
                                <WithStyles(ForwardRef(IconButton))
                                  aria-label="show 17 new notifications"
                                  color="inherit"
                                >
                                  <ForwardRef(IconButton)
                                    aria-label="show 17 new notifications"
                                    classes={
                                      Object {
                                        "colorInherit": "MuiIconButton-colorInherit",
                                        "colorPrimary": "MuiIconButton-colorPrimary",
                                        "colorSecondary": "MuiIconButton-colorSecondary",
                                        "disabled": "Mui-disabled",
                                        "edgeEnd": "MuiIconButton-edgeEnd",
                                        "edgeStart": "MuiIconButton-edgeStart",
                                        "label": "MuiIconButton-label",
                                        "root": "MuiIconButton-root",
                                        "sizeSmall": "MuiIconButton-sizeSmall",
                                      }
                                    }
                                    color="inherit"
                                  >
                                    <WithStyles(ForwardRef(ButtonBase))
                                      aria-label="show 17 new notifications"
                                      centerRipple={true}
                                      className="MuiIconButton-root MuiIconButton-colorInherit"
                                      disabled={false}
                                      focusRipple={true}
                                    >
                                      <ForwardRef(ButtonBase)
                                        aria-label="show 17 new notifications"
                                        centerRipple={true}
                                        className="MuiIconButton-root MuiIconButton-colorInherit"
                                        classes={
                                          Object {
                                            "disabled": "Mui-disabled",
                                            "focusVisible": "Mui-focusVisible",
                                            "root": "MuiButtonBase-root",
                                          }
                                        }
                                        disabled={false}
                                        focusRipple={true}
                                      >
                                        <button
                                          aria-label="show 17 new notifications"
                                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                                          disabled={false}
                                          onBlur={[Function]}
                                          onDragLeave={[Function]}
                                          onFocus={[Function]}
                                          onKeyDown={[Function]}
                                          onKeyUp={[Function]}
                                          onMouseDown={[Function]}
                                          onMouseLeave={[Function]}
                                          onMouseUp={[Function]}
                                          onTouchEnd={[Function]}
                                          onTouchMove={[Function]}
                                          onTouchStart={[Function]}
                                          tabIndex={0}
                                          type="button"
                                        >
                                          <span
                                            className="MuiIconButton-label"
                                          >
                                            <WithStyles(ForwardRef(Badge))
                                              badgeContent={1}
                                              color="secondary"
                                            >
                                              <ForwardRef(Badge)
                                                badgeContent={1}
                                                classes={
                                                  Object {
                                                    "anchorOriginBottomLeftCircle": "MuiBadge-anchorOriginBottomLeftCircle",
                                                    "anchorOriginBottomLeftRectangle": "MuiBadge-anchorOriginBottomLeftRectangle",
                                                    "anchorOriginBottomRightCircle": "MuiBadge-anchorOriginBottomRightCircle",
                                                    "anchorOriginBottomRightRectangle": "MuiBadge-anchorOriginBottomRightRectangle",
                                                    "anchorOriginTopLeftCircle": "MuiBadge-anchorOriginTopLeftCircle",
                                                    "anchorOriginTopLeftRectangle": "MuiBadge-anchorOriginTopLeftRectangle",
                                                    "anchorOriginTopRightCircle": "MuiBadge-anchorOriginTopRightCircle",
                                                    "anchorOriginTopRightRectangle": "MuiBadge-anchorOriginTopRightRectangle",
                                                    "badge": "MuiBadge-badge",
                                                    "colorError": "MuiBadge-colorError",
                                                    "colorPrimary": "MuiBadge-colorPrimary",
                                                    "colorSecondary": "MuiBadge-colorSecondary",
                                                    "dot": "MuiBadge-dot",
                                                    "invisible": "MuiBadge-invisible",
                                                    "root": "MuiBadge-root",
                                                  }
                                                }
                                                color="secondary"
                                              >
                                                <span
                                                  className="MuiBadge-root"
                                                >
                                                  <ForwardRef>
                                                    <WithStyles(ForwardRef(SvgIcon))>
                                                      <ForwardRef(SvgIcon)
                                                        classes={
                                                          Object {
                                                            "colorAction": "MuiSvgIcon-colorAction",
                                                            "colorDisabled": "MuiSvgIcon-colorDisabled",
                                                            "colorError": "MuiSvgIcon-colorError",
                                                            "colorPrimary": "MuiSvgIcon-colorPrimary",
                                                            "colorSecondary": "MuiSvgIcon-colorSecondary",
                                                            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                                            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                                            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                                            "root": "MuiSvgIcon-root",
                                                          }
                                                        }
                                                      >
                                                        <svg
                                                          aria-hidden={true}
                                                          className="MuiSvgIcon-root"
                                                          focusable="false"
                                                          viewBox="0 0 24 24"
                                                        >
                                                          <path
                                                            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                                                          />
                                                        </svg>
                                                      </ForwardRef(SvgIcon)>
                                                    </WithStyles(ForwardRef(SvgIcon))>
                                                  </ForwardRef>
                                                  <span
                                                    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                                                  >
                                                    1
                                                  </span>
                                                </span>
                                              </ForwardRef(Badge)>
                                            </WithStyles(ForwardRef(Badge))>
                                          </span>
                                          <WithStyles(memo)
                                            center={true}
                                          >
                                            <ForwardRef(TouchRipple)
                                              center={true}
                                              classes={
                                                Object {
                                                  "child": "MuiTouchRipple-child",
                                                  "childLeaving": "MuiTouchRipple-childLeaving",
                                                  "childPulsate": "MuiTouchRipple-childPulsate",
                                                  "ripple": "MuiTouchRipple-ripple",
                                                  "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                                  "rippleVisible": "MuiTouchRipple-rippleVisible",
                                                  "root": "MuiTouchRipple-root",
                                                }
                                              }
                                            >
                                              <span
                                                className="MuiTouchRipple-root"
                                              >
                                                <TransitionGroup
                                                  childFactory={[Function]}
                                                  component={null}
                                                  exit={true}
                                                />
                                              </span>
                                            </ForwardRef(TouchRipple)>
                                          </WithStyles(memo)>
                                        </button>
                                      </ForwardRef(ButtonBase)>
                                    </WithStyles(ForwardRef(ButtonBase))>
                                  </ForwardRef(IconButton)>
                                </WithStyles(ForwardRef(IconButton))>
                              </div>
                            </div>
                          </ForwardRef(Toolbar)>
                        </WithStyles(ForwardRef(Toolbar))>
                      </header>
                    </ForwardRef(Paper)>
                  </WithStyles(ForwardRef(Paper))>
                </ForwardRef(AppBar)>
              </WithStyles(ForwardRef(AppBar))>
            </Styled(WithStyles(ForwardRef(AppBar)))>
          </div>
        </TopBar>
      </Connect(TopBar)>
    </Provider>
  </Router>
</BrowserRouter>
`;
